# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../wvdial/glibc.patch
# Copyright (C) 2009 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

--- ./wvmodemscan.cc.orig	2007-07-20 00:22:03.000000000 +0300
+++ ./wvmodemscan.cc	2009-06-20 21:43:58.000000000 +0300
@@ -495,11 +495,16 @@
 	// (no internal ISDN support)   || !strncmp(e->d_name, "ttyI", 4);
 }
 
-
+#if defined(__GLIBC__) && __GLIBC_PREREQ(2, 10)
+static int filesort(const dirent **e1, const dirent **e2)
+#else
 static int filesort(const void *_e1, const void *_e2)
+#endif
 {
+#if !(defined(__GLIBC__) && __GLIBC_PREREQ(2, 10))
     dirent const * const *e1 = (dirent const * const *)_e1;
     dirent const * const *e2 = (dirent const * const *)_e2;
+#endif
     const char *p1, *p2;
     int diff;

